### ISSUE TITLE
Error piping on Conemu

### ISSUE THREAD

**Issue by giggio** [moby/moby#28814]
When you pipe to `docker build` on Windows it fails under Conemu.
It is probably related to [these lines](https://github.com/docker/docker/blob/28cca5dcaa2aa21123805d7f2d51916ea1f38d71/pkg/term/term_windows.go#L74) which explicitly target Conemu.

### Versions

````
C:\> docker version
Client:
 Version:      1.12.3
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   6b644ec
 Built:        Thu Oct 27 00:09:21 2016
 OS/Arch:      windows/amd64
 Experimental: true

Server:
 Version:      1.12.3
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   6b644ec
 Built:        Thu Oct 27 00:09:21 2016
 OS/Arch:      linux/amd64
 Experimental: true
````

ConEmu build: x64
ConEmu version: 16.10.22.0
OS version: Windows 10 x64 (Version 1607 OS Build 14393.447)
Used shell version (Far Manager, git-bash, cmd, powershell, cygwin, whatever): ???

### Problem description

When using Docker for Windows, while using Linux containers, if you run on Conemu:

And PowerShell:

````powershell
 echo 'FROM hello-world' | docker build -t temp -
````

or

````powershell
'FROM hello-world' | docker build -t temp -
````

Or on cmd:

````cmd
echo FROM hello-world | docker build -t temp -
````

You get this error:

````
unable to prepare context: failed to peek context header from STDIN: Incorrect function.
````

The command works on cmd and PowerShell without Conemu.

**Comment by SergeyLukashevich** [comment]
As a workaround, disabling `ConEmu Settings -> Features -> ANSI and xterm sequences` helps for me.

**Comment by takekazuomi** [comment]
+1
+1
```
$ docker version
Client:
 Version:      17.05.0-ce
 API version:  1.29
 Go version:   go1.7.5
 Git commit:   89658be
 Built:        Thu May  4 21:43:09 2017
 OS/Arch:      windows/amd64

Server:
 Version:      17.05.0-ce
 API version:  1.29 (minimum version 1.24)
 Go version:   go1.7.5
 Git commit:   89658be
 Built:        Thu May  4 21:43:09 2017
 OS/Arch:      windows/amd64
 Experimental: true
```

**Comment by friism** [comment]
cc @jhowardmsft

**Comment by edrevo** [comment]
As far as I can tell, the ConEmu author thinks this is a problem with Docker because Docker doesn't behave exactly the same as it does with Powershell/cmd. Would it be reasonable to fix this by removing the `ConEmuANSI` check in https://github.com/moby/moby/blob/28cca5dcaa2aa21123805d7f2d51916ea1f38d71/pkg/term/term_windows.go#L74-L79 and treat it just like cmd?

I would be happy to make and test that change if that is an acceptable fix for this issue.

**Comment by thaJeztah** [comment]
Looks like it's possible to test that theory; the code checks for the `ConEmuANSI` environment variable to be `"ON"` - what happens if you manually change that environment variable to another value; does it work correctly then?

**Comment by edrevo** [comment]
Yes, it works correctly. It is actually the workaround that's specified in https://github.com/Maximus5/ConEmu/issues/958

**Comment by thaJeztah** [comment]
That comment says; disable the _feature_ in ConEmu; in that case, the code on Docker's side will do emulation.

I'm interested if keeping the feature enabled, but overriding the env-var (to make Docker think the feature isn't enabled) does the job

If that works, there may still be something to check for; does it work because Docker is handling emulating, or did it then correctly detect that emulation is not needed.

**Comment by edrevo** [comment]
With the feature enabled, is it impossible to override the env var unfortunately:

```
> $Env:ConEmuANSI="OFF"
> env
[...]
ConEmuANSI=ON
[...]
```

I have also tried setting a system variable ConEmuANSI="OFF", but ConEmu will override it. It looks like ConEmu is doing some work to ensure that the env var refects whether the feature is enabled or not.

I'll try removing the ConEmuANSI check from the `if` and try with the feature enabled in ConEmu.

**Comment by edrevo** [comment]
A bit of an update on this:

I have removed the `if os.Getenv("ConEmuANSI") == "ON" || os.Getenv("ConsoleZVersion") != ""` altogether and that seems to fix the issue:

```powershell
> echo 'FROM hello-world' | .\build\docker-windows-amd64.exe build -t temp -
Sending build context to Docker daemon  2.048kB
Step 1/1 : FROM hello-world
 ---> 2cb0d9787c4d
Successfully built 2cb0d9787c4d
Successfully tagged temp:latest
SECURITY WARNING: You are building a Docker image from Windows against a non-Windows Docker host. All files and directories added to build context will have '-rwxr-xr-x' permissions. It is recommended to double check and reset permissions for sensitive files and directories.
~\git\cli [master ≡ +0 ~1 -0 !]
> echo 'FROM hello-world' | docker build -t temp -
unable to prepare context: failed to peek context header from STDIN: Función incorrecta.
```

I have also verified that removing this block does not produce a regression on https://github.com/moby/moby/issues/25475

For the record, I am using Windows 10 1803 (17134.112) and ConEmu 180626. I have also verified the fix with ConsoleZ 1.18.3.18143

I have looked at some old PRs and I have seen that there have been attempts to enable native VT code processing in recent versions of Windows, but I am not sure if that effort was abandoned. There is a risk that by removing this we are falling back to a different behavior in Windows 10 than in previous Windows versions. Someone with more knowledge of the codebase should check this.

I'll have opened https://github.com/moby/moby/pull/37600
